src/Compilers/CSharp/Portable/Compiler/MethodBodySynthesizer.Lowered.cs

```diff
@@ -277,4 +285,329 @@ internal static BoundBlock ConstructSingleInvocationMethodBody(SyntheticBoundNod
                         : F.Block(F.Return(invocation));
         }
     }
+
+    internal class SynthesizedAsIntValueMethod : SynthesizedGlobalMethodSymbol
+    {
+        private readonly Emit.PEModuleBuilder _module;
+
+        public SynthesizedAsIntValueMethod(Emit.PEModuleBuilder module, PrivateImplementationDetails pi)
+            : base(module.SourceModule, pi,
+                   module.Compilation.GetSpecialType(SpecialType.System_Boolean),
+                   PrivateImplementationDetails.AsIntValueName)
+        {
+            this._module = module;
+            var compilation = _module.Compilation;
+            this.SetParameters(ImmutableArray.Create<ParameterSymbol>(
+                new SynthesizedParameterSymbol(this, compilation.GetSpecialType(SpecialType.System_Object), 0, RefKind.None, "o"),
+                new SynthesizedParameterSymbol(this, compilation.GetSpecialType(SpecialType.System_Int32), 1, RefKind.Out, "value")
+                ));
+        }
+
+        internal override void GenerateMethodBody(TypeCompilationState compilationState, DiagnosticBag diagnostics)
+        {
+            ///// <summary>
+            ///// If o is of an integral numeric type and contains a value that is in the range of an int,
+            ///// return true and set the value parameter to that value.
+            ///// </summary>
```

Comment:
Please reflect this in the comments so that people know how to use these helpers. But I have to say, usually, functions like that tend to assign the same value when they return false and the helpers don't follow this pattern.